File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        

        HashMap<Integer,Integer> table = new HashMap<Integer,Integer>();
        
        for(int i =0;i < nums.length;i++){

                int diff = target - nums[i];
                if(table.containsKey(diff)){

                return new int[] { table.get(diff), i }; 
                // return indices
                }           
                
                table.put(nums[i] , i);
        }
        return new int[] {}; // or throw exception if no solution
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-2.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        L, R = 0 , len(s) - 1
        
        while L < R:
            if s[L] != s[R]:
                return False
            L = L +  1
            R = R - 1
        return True



        
        